CONTRIBUTING.md

```diff
@@ -138,21 +139,37 @@ _Note: If you are not familiar with Python code, don't let yourself get distract
 
 ### Implementation Examples
 
-Have a look at this method: https://github.com/JOJ0/synadm/blob/107d34b38de71d6d21d78141e78a1b19d3dd5379/synadm/cli/user.py#L185
+Have a look at this method: https://github.com/JOJ0/synadm/blob/68749391d6a291d2fac229214f59924189c775ac/synadm/cli/user.py#L358-L369
 
-and this one: https://github.com/JOJ0/synadm/blob/107d34b38de71d6d21d78141e78a1b19d3dd5379/synadm/api.py#L80
+and this one: https://github.com/JOJ0/synadm/blob/68749391d6a291d2fac229214f59924189c775ac/synadm/api.py#L531-L545
 
 That's all it needs to implement command `synadm user details <user_id>`.
 
-And another example, this time using a POST based API endpoint. It implements command `synadm user password <user_id>`. This is the CLI-level method: https://github.com/JOJ0/synadm/blob/0af918fdeee40bc1d3df7b734a46e76bb31148b9/synadm/cli/user.py#L114
+And another example, this time using a POST based API endpoint. It implements command `synadm user password <user_id>`. This is the CLI-level method: https://github.com/JOJ0/synadm/blob/68749391d6a291d2fac229214f59924189c775ac/synadm/cli/user.py#L276-L301
+
+and again it needs a backend method in `api.py`: https://github.com/JOJ0/synadm/blob/68749391d6a291d2fac229214f59924189c775ac/synadm/api.py#L511-L529
+
+
+### Submitting data & URL encoding
+
+Since version 0.42 `synadm` encodes URL's in a central place - the `ApiRequest.query()` function in the `synadm.api` module.
+
+Variables sent as part of the URL are required to be passed to the `query()` method **unaltered**. Do not use f-strings or str.format, let the `query()` method do the sanitizing of the URL.
+
+If we take a look at the `user password` example in above's chapter, we have:
+
+```
+self.query("post", "v1/reset_password/{user_id}", data=data, user_id=user_id)
+```
+
+- `data` is the dictionary we'd like to send in the body of the request.
+- The `user_id` should be passed as part of the URL. We don't do any formatting or URL-encoding on our end, we hand it over as keyword argument to the `query() method` and use a "replacement field" surrounded by curly braces to mark its position in the URL.
```

Comment:
Rewording

```suggestion
- The `data` argument is passed as the request body and expects a Python dictionary.
- The `{user_id}` part of the URL will be replaced by the keyword argument `user_id`. The value of the keyword argument `user_id` will also be URL encoded to ensure [special cases](https://github.com/JOJ0/synadm/issues/96) don't break things.
```

This might be duplicating technical documentation a bit.

Reply:
That is wonderfully rewritten, straight to the point and even includes the "negative" example and why why why we do this.
Love it! Thanks!

Duplicating technical docs you probably mean that we have this written as code comments somewhere around the `query()` method? Right?

I think contributors will appreciate that they find this information right along with a simple coding example in the contribution docs and don't have to look for code in the first place.

Reply:
> Duplicating technical docs you probably mean that we have this written as code comments somewhere around the `query()` method? Right?

I don't remember... maybe?

> I think contributors will appreciate that they find this information right along with a simple code example in the contribution docs and don't have to look for code in the first place.

Yeah I think that's better in terms of experience.